metamorph/plugins/morph_provision.py

```diff
@@ -0,0 +1,279 @@
+        :param metadata_location -- Dictionary of searched metadata in given metadata file
+        """
+        with open(metadata_path) as metadata_fp:
+            metadata = yaml.load(metadata_fp)
+        for metadata_name, location_value in metadata_location.items():
+            try:
+                extracted_data = self.get_metadata_from_location(metadata, location_value, location_value[-1])
+            except KeyError as detail:
+                raise ProvisionException('Unable to find key "{0}" '
+                                         'in given path "{1}"'.format(detail, metadata_location[metadata_name]))
+            self.update_topology_by_metadata(metadata_name, extracted_data)
+
+    @staticmethod
+    def get_correct_metadata_tree(metadata_source, metadata_location):
+        """
+        Method for getting correct metadata tree
+        This method is need if metadata location contain list instead of dictionary.
+
+        :param metadata_source -- Extracted metadata from metadata file
+        :param metadata_location -- list which contains path to metadata
+
+        :returns Dictionary of correct metadata by metadata location
+        """
+        for metadata_tree in metadata_source:
+            if metadata_location[0] in metadata_tree.keys():
+                return metadata_tree
+        raise ProvisionException('Unable to find key "{}" in given metadata file'.format(metadata_location[0]))
+
+    def update_topology_by_metadata(self, metadata_name, metadata_value):
+        """
+        Method for updating topology by metadata name and its value
+
+        :param metadata_name -- topology metadata name
+        :param metadata_value -- metadata name value
+        """
+        if metadata_name in self.res_defs.keys():
+            self.res_defs[metadata_name] = metadata_value
+            if metadata_name != 'networks':
+                self.res_defs[metadata_name] = str(self.res_defs[metadata_name])
+        elif metadata_name in self.resource_groups.keys():
+            self.resource_groups[metadata_name] = str(metadata_value)
+        elif metadata_name in self.provision_topology.keys():
+            self.provision_topology[metadata_name] = str(metadata_value)
+        else:
+            raise ProvisionException('Unable to find given metadata name "{}". '
+                                     'Name does not exists in topology file'.format(metadata_name))
+
+    def get_metadata_from_location(self, metadata_source, metadata_location, metadata_name):
+        """
+        Method for getting metadata value from metadata file by it's path
+
+        :param metadata_source -- extracted metadata from --metadata-file parameter
+        :param metadata_location -- searching metadata location
+        :param metadata_name -- name of metadata searched in given metadata location
+
+        :returns found metadata
+        """
+        if type(metadata_source) is list:
```

Comment:
Just a note:
Generally speaking, I think it's more pythonic to employ duck typing and do something like:
```
if isinstance(metadata_source, list):
```